frontend/app/routes/companies/index.js

```diff
@@ -0,0 +1,14 @@
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+  queryParams: {
+    q: {
+      refreshModel: true,
+      replace: true
+    }
+  },
+
+  model({ q }) {
+    return this.store.query('company', { q });
```

Comment:
Im moment scheint der `company` Endpoint alle der Firmen zugehörigen Models zurückzuliefern, dies könnte auf dem Backend eingeschränkt werden, so dass mindestens bei den Listen aufrufen nur die Firmen Models zurückgegeben werden. Ausserdem kennt JSON:API einen `select` Filter um nur Felder zu selektieren die in einer gewissen Ansicht benötigt werden, dies könnte hier sinnvoll angewandt werden.